cw_bitcoin/lib/bitcoin_wallet.dart

```diff
@@ -143,49 +144,71 @@ abstract class BitcoinWalletBase extends ElectrumWallet with Store {
     final network = walletInfo.network != null
         ? BasedUtxoNetwork.fromName(walletInfo.network!)
         : BitcoinNetwork.mainnet;
-    final snp = await ElectrumWalletSnapshot.load(name, walletInfo.type, password, network);
+
+    final hasKeysFile = await WalletKeysFile.hasKeysFile(name, walletInfo.type);
+
+    ElectrumWalletSnapshot? snp = null;
+
+    try {
+      snp = await ElectrumWalletSnapshot.load(name, walletInfo.type, password, network);
+    } catch (e) {
+      if (!hasKeysFile) rethrow;
+    }
+
+    final WalletKeysData keysData;
+    // Migrate wallet from the old scheme to then new .keys file scheme
+    if (!hasKeysFile) {
+      final newKeysData =
+          WalletKeysData(mnemonic: snp!.mnemonic, xPub: snp.xpub, passphrase: snp.passphrase);
+      WalletKeysFile.createKeysFile(name, walletInfo.type, password, newKeysData);
+      keysData = newKeysData;
+    } else {
+      keysData = await WalletKeysFile.readKeysFile(name, walletInfo.type, password);
+    }
 
     walletInfo.derivationInfo ??= DerivationInfo(
-      derivationType: snp.derivationType ?? DerivationType.electrum,
-      derivationPath: snp.derivationPath,
+      derivationType: snp?.derivationType ?? DerivationType.electrum,
+      derivationPath: snp?.derivationPath,
     );
 
     // set the default if not present:
-    walletInfo.derivationInfo!.derivationPath = snp.derivationPath ?? electrum_path;
-    walletInfo.derivationInfo!.derivationType = snp.derivationType ?? DerivationType.electrum;
+    walletInfo.derivationInfo!.derivationPath = snp?.derivationPath ?? electrum_path;
+    walletInfo.derivationInfo!.derivationType = snp?.derivationType ?? DerivationType.electrum;
```

Comment:
derivation type and path should be in .keys files as they are sensitive also to how wallet would work and generate addresses and sign them

Reply:
True
Will do

Reply:
I'm not sure, because it is stored in WalletInfo as well (which is stored in the database). It does not make sense to override data with the same data from a different source